File: jeong-hyuk/2월11일/짝지어제거하기.py
def solution(s):
    stack = []

    for char in s:
        if stack and stack[-1] == char:
            stack.pop()  # 스택의 가장 위의 문자와 현재 문자가 같으면 제거
        else:
            stack.append(char)  # 아니라면 스택에 추가

    if not stack:  # 모든 문자를 제거했다면 1 반환
        return 1
    else:
        return 0

File: yht0827/2월/2월 11일/짝지어제거하기.java
import java.util.*;

class Solution {
	public int solution(String s) {
		Stack<Character> st = new Stack<Character>();

		for (int i = 0; i < s.length(); i++) {
			if (st.isEmpty()) {
				st.push(s.charAt(i));
			} else {
				if (st.peek() == s.charAt(i)) {
					st.pop();
				} else {
					st.push(s.charAt(i));
				}
			}
		}

		if (st.isEmpty())
			return 1;
		return 0;
	}
}